### PR TITLE
bpo-35899: Fix Enum handling of empty and weird strings

### PR DESCRIPTION
**Pull request by bdbaraban**
Co-authored-by: Maxwell <maxwellpxt@gmail.com>
Co-authored-by: Stéphane Wirtel <stephane@wirtel.be>

https://bugs.python.org/issue35899

### CODE REVIEW

**Review by matrixise — APPROVED** [review]
LGTM

Thank you for your contribution.

**Comment by matrixise** [comment]
@ethanfurman do you want to review/merge this PR?

Have a nice day,

File: Lib/test/test_enum.py [hunk 1]
```diff
@@ -2730,6 +2730,19 @@ def cycle_enum():
         self.assertEqual(256, len(seen), 'too many composite members created')
 
 
+class TestEmptyAndWeirdString(unittest.TestCase):
+
+    def test_empty_string(self):
+        empty_abc = Enum('empty_abc', ('', 'B', 'C'))
```
Comment (ethanfurman) [inline Lib/test/test_enum.py]: This test should fail with a `ValueError` as empty strings for names are not allowed (include check in line 154 in enum.py).

File: Lib/test/test_enum.py [hunk 2]
```diff
@@ -2730,6 +2730,19 @@ def cycle_enum():
...
+        item = getattr(empty_abc, '')
+        self.assertEqual(item.value, 1)
+
+    def test_weird_character_string(self):
+        weird_abc = Enum('weird_abc', ('!', 'B', 'C'))
```
Comment (ethanfurman) [inline Lib/test/test_enum.py]: Use non-latin letters and numbers for this check as those are supported, but exclamation points, etc., are not supported.

**Review by ethanfurman — CHANGES_REQUESTED** [review]
Looking pretty good!  A couple more changes and this should be ready to go.

**Comment by bdbaraban** [comment]
I have made the requested changes; please review again

Thank you, Ethan. I've updated both files accordingly.

File: Misc/NEWS.d/next/Library/2019-02-16-07-11-02.bpo-35899.cjfn5a.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Enum has been fixed to correctly handle empty strings and strings with "weird" characters (ie. '!') without crashing. Original patch contributed by Maxwell. Assisted by Stéphane Wirtel.
```
Comment (ethanfurman) [inline Misc/NEWS.d/next/Library/2019-02-16-07-11-02.bpo-35899.cjfn5a.rst]: Please change the exclamation point and weird characters comment to an alphanumeric unicode comment.

**Review by ethanfurman — CHANGES_REQUESTED** [review]
One more minor change and this PR is good to go!  Nice work!

**Review by ethanfurman — APPROVED** [review]

**Pull request merged**